### PR TITLE
1.9 CVE fix

### PR DESCRIPTION
**Pull request by Bslabe123**
# Description

Fixes for https://github.com/advisories/GHSA-3wx7-46ch-7rq2, https://github.com/advisories/GHSA-vc3p-29h2-gpcp and https://github.com/advisories/GHSA-3wx7-46ch-7rq2

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ianmacclancy** [comment]
Failure in Ci 
https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=c7198ad5-3930-417d-87b6-89b770d592ac

This looks like a go version issue

`Step #10 - "test": Failed to compile test:
Step #10 - "test": 
Step #10 - "test": # golang.org/x/net/http2
Step #10 - "test": /go/pkg/mod/golang.org/x/net@v0.0.0-20211209124913-491a49abca63/http2/transport.go:417:45: undefined: os.ErrDeadlineExceeded
Step #10 - "test": 
Step #10 - "test": Ginkgo ran 2 suites in 39.281033077s
Step #10 - "test": Test Suite Failed
Step #10 - "test": 
Step #10 - "test": [38;5;228mGinkgo 2.0 is coming soon![0m`

/kick

**Comment by inFocus7** [comment]
/kick (A just-in-case, while looking into it some more.)

---

Agree(d) with Ian after I saw [this post](https://github.com/odeke-em/drive/issues/1128).

_Although_ weirdly enough, it looks like gloo 1.10 did this net bump as well [and passed](https://github.com/solo-io/gloo/commit/945225207acb83bce29b2bfe2ed0af63b11ea29a). From what I saw, both use `go1.16.3` (based on searching in their repos), so I'm confused on what the difference would be...

File: Makefile [hunk 1]
```diff
@@ -50,7 +50,7 @@ else
   endif
 endif
 
-ENVOY_GLOO_IMAGE ?= quay.io/solo-io/envoy-gloo:1.20.4-patch1
+ENVOY_GLOO_IMAGE ?= quay.io/solo-io/envoy-gloo:1.21.3-patch1
```
Comment (sam-heilbron) [inline Makefile:53]: Is this necessary?

Comment (Bslabe123) [reply Makefile]: Tried to fix this the same way v1.10.x fixed it, created a patch with CVE fixes so setting the version to that

**Review by sam-heilbron — COMMENTED** [review]
Can we also bump go-utils to pull in the fix for the logging npe if it's affected (https://github.com/solo-io/go-utils/pull/487)

In the original PR that bumped go-utils to pull in the logging handler npe fix (https://github.com/solo-io/gloo/pull/6709), I added logic to one of our kube2e tests to hit the logging endpoint. The idea was that this test will only pass if the request can be made. Since it's recommended in our production docs I think it's valuable that we have automated testing to confirm that it works. Can we add the same check to this branch as well?

**Review by ianmacclancy — APPROVED** [review]
LGTM

**Pull request merged**